prqlc/prqlc/src/semantic/resolver/names.rs

```diff
@@ -16,16 +16,21 @@ use super::Resolver;
 
 impl Resolver<'_> {
     pub(super) fn resolve_ident(&mut self, ident: &Ident) -> Result<Ident, Error> {
-        let mut ident = ident.clone().prepend(self.current_module_path.clone());
-
-        let mut res = self.resolve_ident_core(&ident);
-        for _ in 0..self.current_module_path.len() {
-            if res.is_ok() {
-                break;
+        let mut res = if let Some(default_namespace) = self.default_namespace.clone() {
+            self.resolve_ident_core(ident, Some(&default_namespace))
+        } else {
+            let mut ident = ident.clone().prepend(self.current_module_path.clone());
+
+            let mut res = self.resolve_ident_core(&ident, None);
+            for _ in 0..self.current_module_path.len() {
+                if res.is_ok() {
+                    break;
+                }
+                ident = ident.pop_front().1.unwrap();
+                res = self.resolve_ident_core(&ident, None);
             }
-            ident = ident.pop_front().1.unwrap();
-            res = self.resolve_ident_core(&ident);
-        }
+            res
+        };
```

Comment:
You mean this?

What's the problem here?

Reply:
We get lots of errors — for example: https://github.com/PRQL/prql/actions/runs/8363115491/job/22895344620?pr=4349#step:14:615

Reply:
I was the tests that were wrong.